File: kaihelper/contracts/user_dto.py
"""
User-related DTOs
Data Transfer Objects used for user management operations.
"""

# --- Standard library imports ---
from dataclasses import dataclass, field
from datetime import datetime
from typing import Optional


@dataclass
class UserDTO:
    """
    Represents a public view of a user record.

    Attributes:
        id (int | None): Unique identifier for the user.
        username (str): User's username.
        email (str): User's email address.
        full_name (str | None): Optional user's full name.
        is_active (bool): Indicates whether the user account is active.
        created_at (datetime): Timestamp when the user was created.
        updated_at (datetime): Timestamp when the user was last updated.
        password (str | None): Optional internal field for password (not exposed externally).
    """
    id: Optional[int] = None
    username: str = ""
    email: str = ""
    full_name: Optional[str] = None
    is_active: bool = True
    created_at: datetime = field(default_factory=datetime.utcnow)
    updated_at: datetime = field(default_factory=datetime.utcnow)
    password: Optional[str] = None


@dataclass
class RegisterUserDTO:
    """
    Represents registration data submitted when creating a new account.

    Attributes:
        username (str): Desired username for the new user.
        email (str): User's email address.
        full_name (str | None): Optional full name of the user.
        password (str): Password for authentication.
        confirm_password (str): Password confirmation to ensure match.
        is_active (bool): Defaults to True upon registration.
        created_at (datetime): Timestamp when the user is created.
    """
    username: str
    email: str
    full_name: Optional[str] = None
    password: str = ""
    confirm_password: str = ""
    is_active: bool = True
    created_at: datetime = field(default_factory=datetime.utcnow)


@dataclass
class LoginRequestDTO:
    """
    Represents the credentials provided during user login.

    Attributes:
        username_or_email (str): Username or email for authentication.
        password (str): Password for login verification.
    """
    username_or_email: str
    password: str


@dataclass
class UserProfileDTO:
    """
    Represents a simplified user profile returned by the API.

    Attributes:
        username (str): Username of the user.
        full_name (str | None): Optional user's full name.
        email (str): Email address of the user.
        created_at (datetime): Account creation timestamp.
    """
    username: str
    full_name: Optional[str] = None
    email: str = ""
    created_at: datetime = field(default_factory=datetime.utcnow)

File: kaihelper.mobile/app/src/main/java/nz/yoobee/kaihelper/models/UserDto.kt
package nz.yoobee.kaihelper.models

/**
 * Represents the payload sent when logging in.
 * Backend expects: { "username_or_email": "...", "password": "..." }
 */
data class LoginRequestDTO(
    val username_or_email: String,
    val password: String
)

/**
 * Generic login response wrapper.
 * Matches backend response: { "success": true, "message": "...", "data": { ...user... } }
 */
data class LoginResponseDTO(
    val success: Boolean,
    val message: String,
    val data: UserDTO? = null,   // ✅ changed from "user" → "data" to match backend ResultDTO
    val code: Int? = null        // optional status code
)

/**
 * Represents the data sent during registration.
 */
data class RegisterUserDTO(
    val username: String,
    val email: String,
    val full_name: String,
    val password: String,
    val confirm_password: String
)

/**
 * Represents a user object returned from backend.
 */
data class UserDTO(
    val id: Int? = null,
    val username: String? = null,
    val email: String? = null,
    val full_name: String? = null,
    val is_active: Boolean = true
)
